tests/demo/metal_pio.h

```diff
@@ -0,0 +1,22 @@
+#include <stdlib.h>
+#include <stdint.h>
+#include <metal/compiler.h>
+
+
+struct metal_pio;
+
+struct metal_pio_vtable {
+  void     (*v_pio_write)(uint32_t data, uint32_t enable);
+  uint32_t (*v_pio_read)(void);
```

Comment:
Please indent these two lines, since you have 4-space indentation everywhere else.